File: CustomShare/src/main/java/abhishekti/appyhigh/customshare/ui/CustomShareTextView.java
package abhishekti.appyhigh.customshare.ui;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Typeface;
import android.util.AttributeSet;
import android.widget.TextView;

import androidx.annotation.Nullable;

import abhishekti.appyhigh.customshare.R;

/**
 * Created by Abhishek Tiwari on 24-02-2021.
 */
public class    CustomShareTextView extends androidx.appcompat.widget.AppCompatTextView {

    public CustomShareTextView(Context context) {
        super(context);
        init(null);
    }

    public CustomShareTextView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(attrs);
    }

    public CustomShareTextView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(attrs);
    }

    private void init(@Nullable AttributeSet attrs){
        String fontname = "poppins_medium.ttf";
        if(attrs!=null){
            TypedArray typedArray = getContext().obtainStyledAttributes(attrs, R.styleable.CustomShareTextView);
            fontname = typedArray.getString(R.styleable.CustomShareTextView_typeface);
            typedArray.recycle();
        }

        try{
            Typeface typeface = Typeface.createFromAsset(getContext().getAssets(), "fonts/"+fontname);
            setTypeface(typeface);
        }catch (Exception e){
            e.printStackTrace();
        }
    }
}

File: CustomShare/src/main/java/abhishekti/appyhigh/customshare/ui/CustomShareTextView.kt
package abhishekti.appyhigh.customshare.ui

import abhishekti.appyhigh.customshare.R
import android.content.Context
import android.content.res.TypedArray
import android.graphics.Typeface
import android.util.AttributeSet
import androidx.annotation.Nullable
import androidx.appcompat.widget.AppCompatTextView

/**
 * Created by Abhishek Tiwari on 02-03-2021.
 */
class CustomShareTextView: AppCompatTextView {

    constructor(context: Context): super(context){
        initView(null)
    }

    constructor(context: Context, @Nullable attributeSet: AttributeSet): super(context, attributeSet){
        initView(attributeSet)
    }

    constructor(context: Context, @Nullable attributeSet: AttributeSet, defStyleAttr: Int): super(context, attributeSet, defStyleAttr){
        initView(attributeSet)
    }

    fun initView(attributeSet: AttributeSet?){
        var fontName: String = "poppins_medium.ttf"
        if(attributeSet==null){
            val typedArray: TypedArray = context.obtainStyledAttributes(attributeSet, R.styleable.CustomShareTextView)
            fontName = typedArray.getString(R.styleable.CustomShareTextView_typeface) ?: "poppins_medium.ttf"
            typedArray.recycle()
        }

        try {
            val typeface: Typeface = Typeface.createFromAsset(context.assets, "fonts/${fontName}")
            setTypeface(typeface)
        }catch (e: Exception){
            e.printStackTrace()
        }
    }
}